pkg/types/types.go

```diff
@@ -318,36 +318,38 @@ func areFloatsExactlyEqual(a, b float64) bool {
 	return a == b
 }
 
-func getPointMajorityValue(values []float64, equalityFunc floatEqualityFunc) (majorityValue float64, majorityCount int, err error) {
+func getPointMajorityValue(values []float64, equalityFunc floatEqualityFunc) (majorityValue float64, isMajority bool, err error) {
 	valuesCount := len(values)
 	if valuesCount == 0 {
-		return 0, 0, errors.New("no value for majority voting")
+		return 0, false, errors.New("no value for majority voting")
 	}
 
-	sort.Slice(values, func(i, j int) bool {
-		return values[i] < values[j]
-	})
-
-	majorityCount = 1
-	majorityValue = values[0]
-
-	currentCount := 1
-	currentValue := values[0]
-	for i := 1; i < len(values); i++ {
-		v := values[i]
-		if equalityFunc(v, currentValue) {
-			currentCount++
-			if currentCount > majorityCount {
-				majorityCount = currentCount
-				majorityValue = v
-			}
+	// Boyer–Moore majority vote algorithm
+	i := 0
+	var m float64
+	for _, v := range values {
+		if i == 0 {
+			m = v
+		}
+		if equalityFunc(m, v) {
```

Comment:
this could be changed to `} else if equalityFunc(m, v) {`, less code path to execute.

Reply:
No it's not a bug. `i` should be 1 after the first iteration. As it is here.

Reply:
oh I see, that's clever. but i think by merging the if statement, it still might make a difference on the benchmark numbers. also moving out the `if i > valuesCount/2`. sounds a bit micro-optimisation.

Reply:
Changed.